README.md

```diff
@@ -145,12 +138,80 @@ for a link-local address is auto-detected.
 
 ## Detailed Description of Capabilities
 
-Here are some more detailes on the additional capabilities of mavlink-router.
-
-### Endpoint Configuration
-
-A description of the different endpoint operating modes and their configuration
-parameters can be found in the [examples/config.sample](examples/config.sample).
+To understand how MAVLink Router forwards messages between endpoints, it
+important to know, that a MAVLink message has to have a sender address, but
+only some messages have a target address (e.g. parameter requests). The sender
+and target address consist of a system and component ID each, so one UAV can
+have a flight controller and companion computer with individual component IDs,
+but sharing the same system ID. In the target address, a component ID or 0 is
+used to broadcast to all components on a system, a sysID of 0 broadcasts to all
+systems.
+
+### Endpoints
+MAVLink Router supports three basic types of endpoints: UART, UDP link and TCP
+client. Additionally, it'll act as a TCP server for dynamic clients (if not
+explicitly deactivated).
+
+Endpoint types (see [examples/config.sample](examples/config.sample) for the
+config file format):
+
+  - UART: For telemetry radios or other serial links
+    * Configuration: UART device path/name and baudrate
+    * Behavior: Data is received and sent right away
+  - UDP:
+    * Configuration: Mode (client or server), IP address and port; both unicast
+      and multicast addresses are possible
+    * Behavior in client mode: Data is sent right away, but can only be
+      received on established socket connection
+    * Behavior in server mode: Data is received right away, but can only be
+      sent to source address of the latest connection
+  - TCP Client:
+    * Configuration: Target IP address and port, reconnection interval in case
+      of disconnection
+    * Behavior: Data is received and sent right after the TCP session is
+      established
+
+Defining endpoints:
+
+  - Endpoints are created by one of these methods:
+    * An endpoint is defined in the configuration file
+    * An endpoint is defined by the corresponding command line option
+    * A TCP client has connected to the TCP server port
+  - Endpoint are destoyed, when
+    * A TCP client disconnects from the TCP server port
+    * MAVLink Router is terminated
+    * (This means that UART, UDP and TCP client endpoints are never destroyed
+      during runtime.)
+
+### Message Routing
+In general, each message received on one endpoint is delivered to all other
+endpoints. But there are some rules to reduce the number of "uninteresting"
+messages being delivered to a system.
+
+Routing rules:
+
+  - A message received on one endpoint is delivered to all other endpoints.
```

Comment:
this is only true for broadcast message. A better wording could be: "A message received on one endpoint is delivered to all endpoints in which that target system/component has been seen. If it's a broadcast message, it's delivered to all endpoints. A message is never sent back to the same endpoint it came from".

Reply:
I thought of these rules as a generic rule with multiple exceptions. But since that doesn't seem to be clear everybody, I changed the order and phrasing of the rules.